test/iterative_verifier_test.go

```diff
@@ -95,6 +140,19 @@ func (t *IterativeVerifierTestSuite) TestBeforeCutoverFailuresFailAgainDuringCut
 	t.Require().Equal("verification failed on table: gftest.test_table_1 for pks: 42", result.Message)
 }
 
+func (t *IterativeVerifierTestSuite) TestBeforeCutoverCompressionFailuresFailAgainDuringCutover() {
+	t.InsertCompressedRowInDb(42, testhelpers.TestCompressedData1, t.Ferry.SourceDB)
+	t.InsertCompressedRowInDb(42, testhelpers.TestCompressedData2, t.Ferry.TargetDB)
+
+	err := t.verifier.VerifyBeforeCutover()
+	t.Require().Nil(err)
+
+	result, err := t.verifier.VerifyDuringCutover()
+	t.Require().Nil(err)
+	t.Require().False(result.DataCorrect)
+	t.Require().Equal(fmt.Sprintf("verification failed on table: %s.%s for pks: %s", "gftest", testhelpers.TestCompressedTable1Name, "42"), result.Message)
+}
```

Comment:
Can we possibly have a symmetrical test for the Data3 and Data4 case?

Reply:
Yep!